wire-runtime/src/test/proto-java/com/google/protobuf/FieldDescriptorProto.java

```diff
@@ -408,6 +409,22 @@ public static Type fromValue(int value) {
     public int getValue() {
       return value;
     }
+
+    private static final class ProtoAdapter_Type extends EnumAdapter<Type> {
+      ProtoAdapter_Type() {
+        super(Type.class);
+      }
+
+      @Override
+      public Type decode(ProtoReader reader) throws IOException {
```

Comment:
How about pushing this implementation down into `EnumAdapter` and just adding `public abstract T fromValue(int)` which gets overridden. Then this implementation just becomes:
```java
@Override public Type fromValue(int value) {
  return Type.fromValue(value);
}
```